bindings/java/src/org/sleuthkit/datamodel/AbstractContent.java

```diff
@@ -316,6 +319,22 @@ public BlackboardArtifact newArtifact(int artifactTypeID) throws TskCoreExceptio
 		return db.newBlackboardArtifact(artifactTypeID, objId);
 	}
 
+	@Override
+	public AnalysisResult newAnalysisResult(BlackboardArtifact.Type artifactType, Score score, String conclusion, String configuration, String justification, Collection<BlackboardAttribute> attributesList) throws TskCoreException {
+		
+		CaseDbTransaction trans = db.beginTransaction();
+		try {
+			AnalysisResult result = db.getBlackboard().newAnalysisResult(artifactType, objId, this.getDataSource().getId(), score, conclusion, configuration, justification, attributesList, trans);
+			
+			trans.commit();
+			return result;
+		}
+		catch (BlackboardException ex) {
+			trans.rollback();
+			throw new TskCoreException("Error adding analysis result.", ex);
```

Comment:
A little more debug info, e.g., the object ID of the artifact, might be helpful.